share/completions/portmaster.fish

```diff
@@ -48,8 +48,7 @@ complete -c portmaster -l version --description 'display the version number El E
 # Grab items from the ports directory, max depth 2
 complete -c portmaster -f --description 'Ports Directory' -a "
 (
-        ls -d /usr/ports/(commandline -ct)*/ \
-                | sed -E -e 's#/usr/ports/##' -e 's#/+#/#' -e 's#([^/]+/[^/]+).*#\1#'
+	string match -r '(?<=/usr/ports/)[^/]*(?:/[^/]*)?' (__fish_complete_directories /usr/ports/(commandline -ct))
```

Comment:
Nice use of PCRE features. Note that the `(__fish_complete_directories /usr/ports/(commandline -ct))` does expand past two levels. Your PCRE ensures that only the first two levels are emitted by the completion. It's probably not worthwhile optimizing this any further. I just wanted to make sure you were aware that levels three and higher are expanded by `__fish_complete_directories` before being trimmed to two levels by the `string match`.